lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1331,6 +1332,49 @@ private function getCommitOrder(): array
         return $calc->sort();
     }
 
+    /**
+     * Gets the commit order for deletions.
+     *
+     * @return list<object>
+     */
+    private function getCommitOrderForDeletions(): array
+    {
+        $calc = $this->getCommitOrderCalculator();
+
+        $newNodes = [];
+
+        foreach ($this->entityDeletions as $entity) {
```

Comment:
With `$this->entityDeletions` in here, it wouldn't be needed in `$this->getCommitOrder()` anymore. Otherwise the commit order would calculate a second time using the deletions.